### PR TITLE
[c++] Add `SOMAPointCloudDataFrame` class

### PR DESCRIPTION
**Pull request by jp-dark**
Adds a simple C++ `SOMAPointCloud` class to libtiledbsoma.

### CODE REVIEW

File: libtiledbsoma/test/unit_soma_point_cloud.cc [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+ * @section DESCRIPTION
+ *
+ * This file manages unit tests for the SOMAPointCloud class
+ */
+
+#include "common.h"
+
+const int64_t SOMA_JOINID_DIM_MAX = 99;
+
+TEST_CASE("SOMAPointCloud: basic", "[SOMAPointCloud]") {
```
Comment (johnkerl) [inline libtiledbsoma/test/unit_soma_point_cloud.cc:37]: I'll take it as my own action item to fold in new-shape-related tests for #2407

**Review by johnkerl — CHANGES_REQUESTED** [review]
This looks great -- nice, simple, iterative steps :)

One thing: @mojaveazure had the suggestion (which I second) to rename from `PointCloud` to `PointCloudDataFrame`.

Let's please do this (I can help with search-and-replace on the PRs, happy to!) unless there's a super-solid reason to not to (of which I'm not aware)

I want to make that change before this PR is merged (hence the request-changes).

File: libtiledbsoma/src/soma/soma_point_cloud.h [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+#include "soma_array.h"
+
+namespace tiledbsoma {
+
+class ArrayBuffers;
+
+using namespace tiledb;
+
+class SOMAPointCloud : public SOMAArray {
```
[the remark this answers is not in the document]
Comment (jp-dark) [reply libtiledbsoma/src/soma/soma_point_cloud.h:46]: [sc-56573](https://app.shortcut.com/tiledb-inc/story/56573/spatialdataframe-dataframe-refactor)

**Comment by jp-dark** [comment]
> One thing: @mojaveazure had the suggestion (which I second) to rename from `PointCloud` to `PointCloudDataFrame`.
> 
> Let's please do this (I can help with search-and-replace on the PRs, happy to!) unless there's a super-solid reason to not to (of which I'm not aware)
> 
> I want to make that change before this PR is merged (hence the request-changes).
I don't think all the stakeholders have had a chance to weight in on a name change yet. I'm reaching out to people to make sure there is a consensus before changing the name.

**Review by nguyenv — APPROVED** [review]
lgtm!

**Comment by nguyenv** [comment]
Oh did we want to wait until everyone had consensus on the potential name change before approving and merging?

**Comment by johnkerl** [comment]
> 
Oh did we want to wait until everyone had consensus on the potential name change before approving and merging?

@nguyenv I think so but I also did request-changes on PR to 'hold' that so your approval is fine! :)

**Review by johnkerl — APPROVED** [review]
🌊 🚢 🌊

**Review by nguyenv — APPROVED** [review]

**Pull request merged**